From d4f1a95c8b45a9f70b612614a1bc3008c0d569cf Mon Sep 17 00:00:00 2001
From: Michael Goffioul <michael.goffioul@lincor.com>
Date: Fri, 4 Oct 2019 21:02:49 -0400
Subject: [PATCH 7/7] Add androidboot.hardware to kernel command-line

This replaces the previous use of -DTARGET_PRODUCT=\"$(TARGET_PRODUCT)\"
flag when compiling init.
---
 BoardConfig.mk | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/BoardConfig.mk b/BoardConfig.mk
index 1e6a7bf..8a5f34d 100644
--- a/BoardConfig.mk
+++ b/BoardConfig.mk
@@ -78,7 +78,9 @@ ifneq ($(strip $(BOARD_GPU_DRIVERS)),)
 TARGET_HARDWARE_3D := true
 endif
 
-BOARD_KERNEL_CMDLINE := root=/dev/ram0 androidboot.selinux=permissive$(if $(filter x86_64,$(TARGET_ARCH) $(TARGET_KERNEL_ARCH)),, vmalloc=192M)
+BOARD_KERNEL_CMDLINE := root=/dev/ram0 \
+                        androidboot.selinux=permissive$(if $(filter x86_64,$(TARGET_ARCH) $(TARGET_KERNEL_ARCH)),, vmalloc=192M) \
+                        androidboot.hardware=$(TARGET_PRODUCT)
 TARGET_KERNEL_DIFFCONFIG := device/generic/common/selinux_diffconfig
 
 COMPATIBILITY_ENHANCEMENT_PACKAGE := true
-- 
2.21.0

